tests/rptest/tests/log_segment_ms_test.py

```diff
@@ -197,3 +197,72 @@ def test_segment_timely_rolling_after_change(self):
         wait_until(lambda: self._total_segments_count(topic) > middle_count,
                    timeout_sec=SERVER_HOUSEKEEPING_LOOP * 2,
                    err_msg=f"failed to roll a segment in a timely manner")
+
+    @cluster(num_nodes=3)
+    def test_segment_rolling_with_retention(self):
+        self.redpanda.set_cluster_config({
+            "log_segment_ms": None,
+            "log_segment_ms_min": 10000
+        })
+        topic = TopicSpec(segment_bytes=(1024 * 1024),
+                          replication_factor=1,
+                          partition_count=1)
+        self.client().create_topic(topic)
+
+        producer = VerifiableProducer(context=self.test_context,
+                                      num_nodes=1,
+                                      redpanda=self.redpanda,
+                                      topic=topic.name,
+                                      throughput=10000)
+
+        producer.start()
+        wait_until(
+            lambda: self._total_segments_count(topic) >= 5,
+            timeout_sec=120,
+            err_msg=
+            f"producer failed to produce enough messages to create 5 segments")
```

Comment:
nit: f-string here is not really needed but at the same time I'm ok to keep it, since sometime after a refactor we forget to turn a string to a f-string